asterisk/agi/library/Agi/Action/ServiceAction.php

```diff
@@ -114,4 +117,45 @@ protected function _processGroupPickUp()
 
     }
 
+    protected function _processRecordLocution()
+    {
+        // Local variables to improve readability
+        $service = $this->_service;
+        $caller = $this->_caller;
+
+        // Get Locution id to be recorded
+        $locutionId = substr($this->agi->getExtension(), strlen($service->getCode()) + 1);
```

Comment:
Good, I can do that for the other servicies too. All services have the same format *+code+args but args are different for each one.